tests/unit/s2n_server_psk_extension_test.c

```diff
@@ -0,0 +1,318 @@
+
+            /* Validate that the PSKs not chosen are wiped */
+            for (size_t i = 0; i < conn->psk_params.psk_list.len; i++) {
+                struct s2n_psk *psk = NULL;
+                EXPECT_OK(s2n_array_get(&conn->psk_params.psk_list, i, (void**)&psk));
+                if(i == conn->psk_params.chosen_psk_wire_index) {
+                    EXPECT_NOT_NULL(psk);
+                } else {
+                    EXPECT_EQUAL(psk->identity.data, NULL);
+                    EXPECT_EQUAL(psk->identity.size, 0);
+                    EXPECT_EQUAL(psk->secret.data, NULL);
+                    EXPECT_EQUAL(psk->secret.size, 0);
+                    EXPECT_EQUAL(psk->early_secret.data, NULL);
+                    EXPECT_EQUAL(psk->early_secret.size, 0);
+                }
+            }
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+            EXPECT_SUCCESS(s2n_stuffer_free(&out));
+        }
+    }
+    
+    /* Functional test */
+    {
+        struct s2n_stuffer out = { 0 };
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&out, 0));
+
+        struct s2n_connection *server_conn;
+        struct s2n_connection *client_conn;
+        EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_SERVER));
+        EXPECT_NOT_NULL(client_conn = s2n_connection_new(S2N_CLIENT));
+
+        server_conn->actual_protocol_version = S2N_TLS13;
+        client_conn->actual_protocol_version = S2N_TLS13;
+
+        for (size_t i = 0; i < s2n_array_len(test_cases); i++) {
+            struct s2n_psk *psk = NULL;
+            EXPECT_OK(s2n_array_pushback(&client_conn->psk_params.psk_list, (void**) &psk));
+            EXPECT_OK(s2n_array_pushback(&server_conn->psk_params.psk_list, (void**) &psk));
+            EXPECT_SUCCESS(s2n_psk_init(psk, S2N_PSK_TYPE_EXTERNAL));
+            EXPECT_SUCCESS(s2n_psk_new_identity(psk, test_cases[i].identity, test_cases[i].identity_size));
+            psk->hmac_alg = test_cases[i].hmac_alg;
+        }
+
+        client_conn->psk_params.chosen_psk_wire_index = 0;
+
+        EXPECT_SUCCESS(s2n_server_psk_extension.send(client_conn, &out));
+        EXPECT_SUCCESS(s2n_server_psk_extension.recv(server_conn, &out));
+
+        EXPECT_NOT_NULL(server_conn->psk_params.chosen_psk);
+
+        /* Validate that the PSKs not chosen are wiped */
+        for (size_t i = 0; i < server_conn->psk_params.psk_list.len; i++) {
+            struct s2n_psk *psk = NULL;
+            EXPECT_OK(s2n_array_get(&server_conn->psk_params.psk_list, i, (void**)&psk));
+            if(i == server_conn->psk_params.chosen_psk_wire_index) {
+                EXPECT_NOT_NULL(psk);
+            } else {
+                EXPECT_EQUAL(psk->identity.data, NULL);
+                EXPECT_EQUAL(psk->identity.size, 0);
+                EXPECT_EQUAL(psk->secret.data, NULL);
+                EXPECT_EQUAL(psk->secret.size, 0);
+                EXPECT_EQUAL(psk->early_secret.data, NULL);
+                EXPECT_EQUAL(psk->early_secret.size, 0);
```

Comment:
If you're going to check this in multiple tests, you might want a helper function for it.